runtime/util_core/j9shchelp.c

```diff
@@ -373,3 +373,13 @@ isCacheFileName(J9PortLibrary* portlib, const char* nameToTest, uintptr_t expect
 	}
 	return 0;	
 }
+
+/* Return the OpenJ9 SHA */
+uint64_t
+getOpenJ9Sha()
+{
+	uint64_t sha = 0;
+	char *str = J9VM_VERSION_STRING;
+	scan_hex_u64(&str, &sha);
```

Comment:
There should be checks (or a test) to ensure getting the sha is working. If scan_hex_u64() returns the number of bits, the code could check there are at least 28 bits and the sha value is non-zero, otherwise assert.

How does j9version.h get #included in j9shchelp.c, I don't see it included by any .h file.

Reply:
and note we don't want j9version.h included by a .h file as this isn't good for incremental builds.